modules/dev-tools/deprecation.php

```diff
@@ -199,25 +199,189 @@ private function check_deprecation( $entity, $version, $replacement, $base_versi
 			throw new \Exception( 'Invalid deprecation diff.' );
 		}
 
-		$print_deprecated = false;
+		$print_deprecated = array();
 
-		if ( defined( 'WP_DEBUG' ) && WP_DEBUG && $diff <= self::SOFT_VERSIONS_COUNT ) {
-			// Soft deprecated.
-			if ( ! isset( $this->soft_deprecated_notices[ $entity ] ) ) {
-				$this->soft_deprecated_notices[ $entity ] = [
-					$version,
-					$replacement,
-				];
+		if ( $this->should_print_deprecated( $version, $base_version ) || $this->should_console_log_deprecated() ) {
+			$backtrace = debug_backtrace( DEBUG_BACKTRACE_IGNORE_ARGS );
+
+			$external_sources = $this->find_third_party_sources( $backtrace );
+
+			$source_message = 'Elementor';
+			$plugin_name = 'unknown';
+			if ( ! ( empty( $external_sources ) ) ) {
+				$calling_source = $external_sources[ array_key_first( $external_sources ) ];
+				$plugin_name = $this->get_plugin_name( $calling_source['file'] );
+				if ( empty( $plugin_name ) ) {
+					$plugin_name = 'none';
+				}
+				$source_message = sprintf( '%s on file %s:%d.', $calling_source['function'], $calling_source['file'], $calling_source['line'] );
+			}
+			if ( $this->should_console_log_deprecated() ) {
+				// Soft deprecated.
+				if ( ! isset( $this->soft_deprecated_notices[ $entity ] ) ) {
+					$this->soft_deprecated_notices[ $entity ] = [
+						$version,
+						$replacement,
+						$source_message,
+					];
+				}
 			}
 
-			if ( defined( 'ELEMENTOR_DEBUG' ) && ELEMENTOR_DEBUG ) {
-				$print_deprecated = true;
+			if ( $this->should_print_deprecated( $version, $base_version ) ) {
+				$print_deprecated = array(
+					'source' => $source_message,
+					'plugin' => $plugin_name,
+				);
 			}
 		}
-
 		return $print_deprecated;
 	}
 
+	private function get_plugin_name( $filename ) {
+		$plugin_path = substr( strstr( $filename, WP_PLUGIN_DIR ), strlen( WP_PLUGIN_DIR ) );
```

Comment:
It can be a `theme`, or `mu-plugin` so try to work with `WP_CONTENT_DIR`

Reply:
i tried, it didn't catch the plufins

Reply:
https://github.com/elementor/elementor/pull/22062/files#diff-d2f15ad4308ce3f20df282bb3817b49c5f24aef0dddf8f818a791e8bea3d92bdR401